cylc/flow/scripts/set.py

```diff
@@ -341,11 +351,44 @@ def validate_opts(output_opt: List[str], prereq_opt: List[str]):
         raise InputError("Use --prerequisite or --output, not both.")
 
 
+def validate_tokens(tokens_list):
+    """Check the cycles/tasks provided.
+
+    This checks that cycle/task selectors have not been provided in the IDs.
+
+    Examples:
+        Good:
+        >>> validate_tokens([Tokens('w//c')])
+        >>> validate_tokens([Tokens('w//c/t')])
+
+        Bad:
+        >>> validate_tokens([Tokens('w//c:s')])
+        Traceback (most recent call last):
+        cylc.flow.exceptions.InputError
+        >>> validate_tokens([Tokens('w//c/t:s')])
+        Traceback (most recent call last):
+        cylc.flow.exceptions.InputError
+
+    """
+    for tokens in tokens_list:
+        if tokens['cycle_sel']:
+            raise InputError(SELECTOR_ERROR.format(
+                tokens['cycle'],
+                tokens['cycle_sel'],
+            ))
+        if tokens['task_sel']:
+            raise InputError(SELECTOR_ERROR.format(
+                tokens['task'],
+                tokens['task_sel'],
+            ))
```

Comment:
I absent-mindedly tried running a command like this `cylc set workflow//cycle/task:output` rather than `cylc set workflow//cycle/task --out=output`.

The CLI accepted this, but it didn't match any tasks because it merged the output in with the task name.